File: app/src/main/java/com/musicplayer/OpenMusic/data/base/UUIDConverter.java
package com.musicplayer.OpenMusic.data.base;

import androidx.annotation.Nullable;
import androidx.room.TypeConverter;

import java.util.UUID;

public class UUIDConverter {

    @TypeConverter
    @Nullable
    public static String fromUUID(@Nullable UUID uuid) {
        return uuid == null ? null : uuid.toString();
    }

    @TypeConverter
    @Nullable
    public static UUID uuidFromString(@Nullable String string) {
        return string == null ? null : UUID.fromString(string);
    }
}

File: app/src/main/kotlin/com/musicplayer/openmusic/data/base/UUIDConverter.kt
package com.musicplayer.openmusic.data.base

import androidx.room.TypeConverter
import java.util.*

object UUIDConverter {
    @JvmStatic
    @TypeConverter
    fun fromUUID(uuid: UUID): String {
        return uuid.toString()
    }

    @JvmStatic
    @TypeConverter
    fun uuidFromString(string: String): UUID {
        return UUID.fromString(string)
    }
}